### PR TITLE
add support for XIM

### PR DESCRIPTION
**Pull request by Determinant**
This provides with a tentative implementation to add support for XIM. The major modifications are made to winit so that the alacritty side is minimal. Temporarily redirect the glutin repo to my branch because we want glutin to use my current version of winit (https://github.com/tomaka/winit/pull/223). This directly addresses issue https://github.com/jwilm/alacritty/issues/44. It would be better to open another PR to change @jwilm own  glutin repo to aggregate the changes.

### CODE REVIEW

File: src/window.rs [hunk 1]
```diff
@@ -184,6 +184,21 @@ impl Window {
         title: &str
     ) -> Result<Window> {
         let event_loop = EventsLoop::new();
+
+        if cfg!(target_os = "linux") {
```
Comment (quininer) [inline src/window.rs]: It may work in `freebsd`/`openbsd`?

Comment (Determinant) [inline src/window.rs]: Indeed. Lemme make this consistent with winit tomorrow.

**Comment by jwilm** [comment]
if you want to update this PR to be against master, it should have the latest winit.

**Comment by Determinant** [comment]
@jwilm Must be drunk at the time of writing last comment (misread your comment), Lol. Working on the PR against master now.

**Pull request closed without merging**